timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -219,18 +219,21 @@ export class TimeGraphChart extends TimeGraphChartLayer {
             update
         )) {
             this.fetching = true;
-            const rowData = await this.providers.dataProvider(viewRange, resolution);
-            if (rowData) {
-                this.providedResolution = rowData.resolution;
-                this.providedRange = rowData.range;
-                this.setRowModel(rowData.rows);
-                this.removeChildren();
-                this.addRows(this.rows, this.rowController.rowHeight);
-                if (this.isNavigating) {
-                    this.selectStateInNavigation();
+            try {
+                const rowData = await this.providers.dataProvider(viewRange, resolution);
+                if (rowData) {
+                    this.providedResolution = rowData.resolution;
+                    this.providedRange = rowData.range;
+                    this.setRowModel(rowData.rows);
+                    this.removeChildren();
+                    this.addRows(this.rows, this.rowController.rowHeight);
+                    if (this.isNavigating) {
+                        this.selectStateInNavigation();
+                    }
                 }
+            } finally {
+                this.fetching = false;
```

Comment:
though not related, can we also set this.isNavigating to 'false' to make sure we set all the flags correctly.

Reply:
make sense